data/src/emulator.js

```diff
@@ -240,6 +240,16 @@ class EmulatorJS {
         this.canvas.classList.add('ejs_canvas');
         this.videoRotation = ([0, 1, 2, 3].includes(this.config.videoRotation)) ? this.config.videoRotation : this.preGetSetting("videoRotation") || 0;
         this.videoRotationChanged = false;
+        this.config.screenshot = this.config.screenshot || {};
+        this.config.screenshot.source = (this.config.screenshot.source === "canvas" || this.config.screenshot.source === "retroarch") ? this.config.screenshot.source : "canvas";
+        this.config.screenshot.format = (typeof this.config.screenshot.format === "string") ? this.config.screenshot.format : "png";
+        this.config.screenshot.quality = (typeof this.config.screenshot.quality === "number") ? this.config.screenshot.quality : 1;
+        this.config.screenRecording = this.config.screenRecording || {};
+        this.config.screenRecording.format = (typeof this.config.screenRecording.format === "string") ? this.config.screenRecording.format : "detect";
+        this.config.screenRecording.quality = (typeof this.config.screenRecording.quality === "number") ? this.config.screenRecording.quality : 1;
+        this.config.screenRecording.fps = (typeof this.config.screenRecording.fps === "number") ? this.config.screenRecording.fps : 30;
+        this.config.screenRecording.videoBitrate = (typeof this.config.screenRecording.videoBitrate === "number") ? this.config.screenRecording.videoBitrate : 2.5 * 1024 * 1024;
+        this.config.screenRecording.audioBitrate = (typeof this.config.screenRecording.audioBitrate === "number") ? this.config.screenRecording.audioBitrate : 192 * 1024;
```

Comment:
I prefer to treat `this.config` as immutable - Could we make this a seperate property, something like `this.capture.video`/`this.capture.photo`?